### ISSUE TITLE
Implement unimplemented features on /publicRooms

### ISSUE THREAD

**Issue by babolivier** [matrix-org/dendrite#638]
- [ ] Support a `server` query parameter to specify which server the user wants the room directory of.
- [x] Implement [POST /_matrix/client/r0/publicRooms](https://matrix.org/docs/spec/client_server/r0.4.0.html#post-matrix-client-r0-publicrooms)

While I'm not sure about the difficulty of the former, the latter should be pretty simple as it only requires to reuse the logic implemented for the `GET` one.

**Comment by anoadragon453** [comment]
I believe the second bullet point may already be done. See #743.

**Comment by anstue** [comment]
Hey, what do you mean by server parameter? I can't find it in the specs: https://matrix.org/docs/spec/server_server/unstable#post-matrix-federation-v1-publicrooms

**Comment by masozzi** [comment]
@anstue 
> Hey, what do you mean by server parameter? I can't find it in the specs:
> https://matrix.org/docs/spec/server_server/unstable#post-matrix-federation-v1-publicrooms

You are looking at the server-server documentation, you should look into the client-server docs:
https://matrix.org/docs/spec/client_server/r0.6.0#post-matrix-client-r0-publicrooms

**Comment by prateek2211** [comment]
i would like to work on first part

**Comment by rreuvekamp** [comment]
Is this issue still relevant after #1174 ?

**Comment by Kegsay** [comment]
Superseded by https://github.com/matrix-org/dendrite/issues/1301
